hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/RecordLevelIndexSupport.scala

```diff
@@ -229,6 +229,80 @@ object RecordLevelIndexSupport {
     }
   }
 
+  def fetchQueryWithAttribute(queryFilter: Expression, recordKeyOpt: Option[String], literalGenerator: Function2[AttributeReference, Literal, String],
```

Comment:
possible to add a UT for this? I know the test already covers both kinds of expressions, but would be good to have a separate UT just for this util.

Reply:
Created https://issues.apache.org/jira/browse/HUDI-8649